### PR TITLE
Template Refactor

### PR DESCRIPTION
**Pull request by kinrezC**
- Adds example agents for counter and price path simulator runs
- Adds updated configuration technique with dynamic configs
- Adds `clap` so that user can specify simulation config at runtime
- Uses updated `EventLogger` api

### CODE REVIEW

File: .vscode/settings.json
Comment (0xJepsen) [inline .vscode/settings.json:1]: I think we can ignore this?

**Review by 0xJepsen — APPROVED** [review]
looks good to me

File: .vscode/settings.json
Comment (kinrezC) [reply .vscode/settings.json:1]: ill just delete it

**Review by Autoparallel — APPROVED** [review]
nice!

**Pull request merged**